dissect/target/plugins/os/unix/modules.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass
+from typing import Iterator, Optional
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export, internal
+from dissect.target.target import Target
+
+ModuleRecord = TargetRecordDescriptor(
+    "unix/module",
+    [
+        ("string", "name"),
+        ("varint", "size"),
+        ("varint", "refcount"),
+        ("string[]", "used_by"),
+        ("path", "source"),
+    ],
+)
+
+
+@dataclass
+class Module:
+    path: str
+    name: str
+    size: int
+    refcnt: int
+    used_by: list[str]
+
+
+class ModulePlugin(Plugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._module_paths = list(self.target.fs.path("/sys/module").iterdir())
+
+    def check_compatible(self) -> bool:
+        return len(self._module_paths) > 0
+
+    @internal
```

Comment:
```suggestion
```
The `@internal` decorator is only necessary if you wish to expose a function to the Python API (`t._iterate_modules` in this case) but not be usable by `target-query`.